### PR TITLE
dockerfile: use xx-info march when downloading dockerd

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/buildx/pull/1897#discussion_r1254948724

### CODE REVIEW

**Review by tonistiigi — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -28,16 +28,7 @@ FROM gobase AS docker
 ARG TARGETPLATFORM
 ARG DOCKER_VERSION
 WORKDIR /opt/docker
-RUN DOCKER_ARCH=$(case ${TARGETPLATFORM:-linux/amd64} in \
-    "linux/amd64")   echo "x86_64"  ;; \
-    "linux/arm/v6")  echo "armel"   ;; \
```
Comment (tonistiigi) [inline Dockerfile:33]: Actually this doesn't look correct for arm/vX . https://github.com/tonistiigi/xx/blob/master/base/xx-info#L211 . I'm not sure if this is based on anything? Arm is like debian but aarch64/ppc64le are not debian 😕

Comment (crazy-max) [reply Dockerfile:33]: Yes indeed, I recall we had a discussion with @tianon and @thaJeztah to align them in the future.

Comment (tonistiigi) [reply Dockerfile:33]: In that case, we could still do a `case` in `xx-info` with fewer checks than the original, or if you think it's not worth it then we can close this and keep as is.

Comment (crazy-max) [reply Dockerfile:33]: Yes let's keep it as is. I'm closing this one and mark it as needs/follow-up.

**Pull request closed without merging**